keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1063 @@
+  spec.
+
+### Test Plan
+
+The admission controller can safely be enabled as a no-op with the default-defaults, i.e. everything
+is privileged. This will let us run the admission controller in our standard E2E test jobs, by
+relabeling specific test namespaces.
+
+**E2E Tests:** The following tests should be added:
+
+1. Enforce mode tests:
+    - Test all profile levels
+    - Test profile version support
+2. Warning mode tests:
+    - Profile levels & version support
+3. Namespace policy relabeling
+    - Ensure labeling completes even when there are warnings
+    - Test warning on violating pods
+    - Test dry-run mode
+
+Additionally, we should add tests to the upgrade test suite to ensure that version skew is properly
+handled:
+
+- A minimally specified pod (just a container image) should always be allowed by the baseline
+  policy.
+- A privileged pod should never be allowed by baseline or restricted
+- A Fully specified pod within the bounds of baseline should be allowed by baseline, and rejected by
+  restricted.
+- A minimally specified restricted pod should be allowed at a pinned version.
+
+**Integration Tests:** Audit mode tests should be added to integration testing, where we have
+existing audit logging tests.
+
+**Manual Testing Resources:** Pod resources will be provided covering all dimensions of the baseline
+& restricted profiles, for validation of 3rd party policy implementations. These have been drafted
+by @JimBugwadia: https://github.com/JimBugwadia/pod-security-tests
+
+**Unit Tests:** Both the library and admission controller implementations will have thorough
+coverage of unit tests.
+
+### Monitoring
+
+A single metric will be added to track policy evaluations against pods and [templated pods].
+[Namespace evaluations](#namespace-policy-update-warnings) are not counted.
+
+```
+<component_name>_evaluations_total
+```
+
+The metric will use the following labels:
+
+1. `decision {allow, deny, exempt, error}` - The policy decision. Error is reserved for panics or
+   other errors in policy evaluation. Update requests that are out of scope (see [Updates](#updates)
+   above) are not counted.
+3. `policy_level {privileged, baseline, restricted}` - The policy level that the request was
+   evaluated against.
+4. `policy_version {latest, v1.YY, >v1.ZZ}` - The policy version that was used for the evaluation.
+   How to constrain cardinality is unresolved (see below).
+5. `mode {enforce, warn, audit}` - The type of evaluation mode being recorded. Note that a single
+   request can increment this metric 3 times, once for each mode. If this admission controller is
+   enabled, every every create request and in-scope update request will at least increment the
+   `enforce` total.
+6. `request_operation {create, update}` - The operation of the request being checked.
+
+<<[UNRESOLVED]>>
+
+_Non-blocking: can be decided on the implementing PR_
```

Comment:
Same q: Just want to confirm this will be updated once implemented -yes?

Reply:
Yep!